Accordion.js

```diff
@@ -0,0 +1,474 @@
+			"-ms-animation": "MSAnimationEnd" // IE 10
+		};
+		// NOTE: returns null if event is not supported
+		var fakeElement = document.createElement("fakeElement");
+		for (var event in animationEndEvents) {
+			if (fakeElement.style[event] !== undefined) {
+				return animationEndEvents[event];
+			}
+		}
+		return null;
+	})();
+
+	function listenAnimationEndEvent(element) {
+		if (animationEndEvent) {
+			return new Promise(function (resolve) {
+				var handler = element.on(animationEndEvent, function () {
+					handler.remove();
+					resolve();
+				});
+			});
+		}
+	}
+
+	/**
+	 * A layout container that display a vertically stacked list of Panels whose titles are all visible, but only one
+	 * or at least one panel's content is visible at a time (depending on the singleOpen property value).
+	 *
+	 * Once the panels are in an accordion, they become collapsible Panels by replacing their headers by ToggleButtons.
+	 *
+	 * When a panel is open, it fills all the available space with its content.
+	 *
+	 * @example:
+	 * <d-accordion id="accordion" selectedChildId="panel1">
+	 *     <d-panel id="panel1">...</d-panel>
+	 *     <d-panel id="panel2">...</d-panel>
+	 *     <d-panel id="panel3">...</d-panel>
+	 * </d-accordion>
+	 * @class module:deliteful/Accordion
+	 * @augments module:delite/DisplayContainer
+	 */
+	var Accordion = dcl([DisplayContainer, KeyNav], /** @lends module:deliteful/Accordion# */ {
+
+		/**
+		 * The name of the CSS class of this widget.
+		 * @member {string}
+		 * @default "d-accordion"
+		 */
+		baseClass: "d-accordion",
+
+		/**
+		 * The id of the panel to be open at initialization.
+		 * If not specified, the default open panel is the first one.
+		 * @member {string}
+		 * @default ""
+		 */
+		selectedChildId: "",
+
+		/**
+		 * If true, only one panel is open at a time.
+		 * If false, several panels can be open at a time, but there's always at least one open.
+		 * @member {boolean}
+		 * @default true
+		 */
+		singleOpen: true,
+
+		/**
+		 * If true, animation is used when a panel is opened or closed.
+		 * @member {boolean}
+		 * @default true
+		 */
+		animate: true,
+
+		/**
+		 * The default CSS class to apply to DOMNode in children headers to make them display an icon when they are
+		 * open. If a child panel has its own iconClass specified, that value is used on that panel.
+		 * @member {string}
+		 * @default ""
+		 */
+		openIconClass: "",
+
+		/**
+		 * The default CSS class to apply to DOMNode in children headers to make them display an icon when they are
+		 * closed. If a child panel has its own closedIconClass specified, that value is used on that panel.
+		 * @member {string}
+		 * @default ""
+		 */
+		closedIconClass: "",
+
+		_panelList: [],
+
+		_numOpenPanels: 0,
+
+		_changeHandler: function (event) {
+			var panel = event.target.parentNode;
+			//Case when the event is fired by the label or the icon
```

Comment:
I think we tend to have spaces between // and comments. Might be good for consistency to do the same.